controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -3399,5 +3808,205 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
         };
       });
     },
+
+    createOIDCProvider: (req, ctx) => {
+      const logger = opts.logger.child({
+        service: ctx.service.typeName,
+        method: ctx.method.name,
+      });
+
+      return handleError<PlainMessage<CreateOIDCProviderResponse>>(logger, async () => {
+        const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
+        const orgRepo = new OrganizationRepository(opts.db);
+        const userRepo = new UserRepository(opts.db);
+
+        const hasAccess = await userRepo.checkUserAccess({
```

Comment:
Only admins should be able to create new providers.

Reply:
Done